contrib/platform/src/com/sun/jna/platform/win32/Psapi.java

```diff
@@ -266,6 +266,46 @@
      */
     boolean GetPerformanceInfo(PERFORMANCE_INFORMATION pPerformanceInformation, int cb);
 
+    /**
+     * It is a good idea to use a large array, because it is hard to predict
+     * how many processes there will be at the time you call EnumProcesses.
+     * <br>
+     * To determine how many processes were enumerated, divide the
+     * pBytesReturned value by sizeof(DWORD). There is no indication given
+     * when the buffer is too small to store all process identifiers.
+     * Therefore, if pBytesReturned equals cb, consider retrying the call
```

Comment:
You've done this in the test case, but it's a good idea not to leave this to the user.  Can you add a new `PsapiUtil` class with a method that implements this retry logic like you are doing in the test case?  Then you can copy over only the correct number of values from the int array and return that to the user.  Then users only have to call the util method.

Reply:
Sure!